spring-xd-dirt/src/main/java/org/springframework/xd/dirt/rest/StreamsController.java

```diff
@@ -67,4 +75,27 @@ public StreamsController(StreamDeployer streamDeployer, StreamDefinitionReposito
 	protected StreamDefinition createDefinition(String name, String definition) {
 		return new StreamDefinition(name, definition);
 	}
+
+	@ResponseBody
+	@RequestMapping(value = "/clean/{transport}/{stream}")
+	@ResponseStatus(HttpStatus.OK)
+	public List<String> clean(@PathVariable String transport, @PathVariable String stream,
+			@RequestParam(required = false) String adminUri,
+			@RequestParam(required = false) String user,
+			@RequestParam(required = false) String pw,
+			@RequestParam(required = false) String vhost,
+			@RequestParam(required = false) String busPrefix) {
+		Assert.isTrue("rabbit".equals(transport), "Only rabbit transports can be cleaned");
+		String adminUri2 = adminUri == null ? "http://localhost:15672" : adminUri;
```

Comment:
All those defaults should not reside in the Controller layer, but rather _e.g._ in the cleaner itself